### PR TITLE
[M2-3055] Feature: Toast

### PR DESCRIPTION
**Pull request by moiskillnadne**

### CODE REVIEW

File: src/entities/activity/model/hooks/useAnswerSubmittedToast.ts [hunk 1]
```diff
@@ -0,0 +1,24 @@
...
+export const useAnswerSubmittedToast = (props: Props) => {
+  const location = useLocation()
+  const { greaterThanSM } = useCustomMediaQuery()
+
+  useEffect(() => {
+    const isAnswersSubmitted = location.state?.isAnswersSubmitted || false
+
+    if (isAnswersSubmitted) {
+      toast.success(props.text, { position: greaterThanSM ? "bottom-left" : "bottom-center" })
+      location.state = {}
```
Comment (BamMironov) [inline src/entities/activity/model/hooks/useAnswerSubmittedToast.ts:21]: The hook responsible for displaying toasts must not be responsible for doing other things

File: src/entities/activity/model/hooks/useAnswerSubmittedToast.ts [hunk 2]
```diff
@@ -0,0 +1,24 @@
...
+type Props = {
+  text: string
+}
+
+export const useAnswerSubmittedToast = (props: Props) => {
+  const location = useLocation()
+  const { greaterThanSM } = useCustomMediaQuery()
+
+  useEffect(() => {
+    const isAnswersSubmitted = location.state?.isAnswersSubmitted || false
```
Comment (BamMironov) [inline src/entities/activity/model/hooks/useAnswerSubmittedToast.ts:17]: Well, generally not a reliable source to subscribe to. Better to render toast on more concrete events like function callbacks

File: src/shared/ui/Toast/style.scss [hunk 3]
```diff
@@ -0,0 +1,6 @@
+@media screen and (max-width: 480px) {
+  .Toastify__toast-container--bottom-center {
+    bottom: 1rem;
```
Comment (BamMironov) [inline src/shared/ui/Toast/style.scss:3]: Why rem?

File: src/shared/ui/Toast/style.scss [hunk 4]
```diff
@@ -0,0 +1,6 @@
...
+    transform: translateX(12%);
```
Comment (BamMironov) [inline src/shared/ui/Toast/style.scss:4]: Why translateX?

File: src/widgets/ActivityDetails/model/hooks/useAnswers.ts [hunk 5]
```diff
@@ -124,9 +132,10 @@ export const useAnswer = (props: UseAnswerProps) => {
       getGroupInProgressByIds,
       getSubmitId,
       props.activityId,
-      props.appletEncryption,
-      props.appletId,
-      props.appletVersion,
+      props.appletDetails.activityFlows,
+      props.appletDetails.encryption,
+      props.appletDetails.id,
+      props.appletDetails.version,
```
Comment (BamMironov) [inline src/widgets/ActivityDetails/model/hooks/useAnswers.ts:138]: Does it make any sense to wrap processAnswers in useCallback?
If so then you can provide just  props.appletDetails as a dependency

File: src/widgets/ActivityDetails/model/hooks/useEntityComplete.ts [hunk 6]
```diff
@@ -19,21 +19,35 @@ export const useEntityComplete = (props: Props) => {
...
+  const completeEntityAndRedirect = () => {
     entityCompleted({
       appletId: props.appletDetails.id,
       entityId: props.flowId ? props.flowId : props.activityId,
       eventId: props.eventId,
     })
 
+    const entityCompletedState = {
+      isAnswersSubmitted: true,
+    }
```
Comment (BamMironov) [inline src/widgets/ActivityDetails/model/hooks/useEntityComplete.ts:31]: I see the idea. If it is just used to render a toast then we should thing of a different idea as this one violates SOLID

**Review by BamMironov — CHANGES_REQUESTED** [review]

File: src/widgets/ActivityDetails/model/hooks/useAnswers.ts (hunk 5 above)
Comment (moiskillnadne) [reply src/widgets/ActivityDetails/model/hooks/useAnswers.ts:138]: Done

File: src/entities/activity/model/hooks/useAnswerSubmittedToast.ts (hunk 2 above)
Comment (moiskillnadne) [reply src/entities/activity/model/hooks/useAnswerSubmittedToast.ts:17]: Done

File: src/widgets/ActivityDetails/model/hooks/useEntityComplete.ts (hunk 6 above)
Comment (moiskillnadne) [reply src/widgets/ActivityDetails/model/hooks/useEntityComplete.ts:31]: Done

File: src/shared/ui/Toast/style.scss (hunk 3 above)
Comment (moiskillnadne) [reply src/shared/ui/Toast/style.scss:3]: Removed at all

File: src/shared/ui/Toast/style.scss (hunk 4 above)
Comment (moiskillnadne) [reply src/shared/ui/Toast/style.scss:4]: Removed at all

File: src/entities/activity/model/hooks/useAnswerSubmittedToast.ts (hunk 1 above)
Comment (moiskillnadne) [reply src/entities/activity/model/hooks/useAnswerSubmittedToast.ts:21]: Done

File: src/entities/activity/model/hooks/useAnswerSubmittedToast.ts [hunk 7]
```diff
@@ -1,24 +1,9 @@
...
-  text: string
-}
-
-export const useAnswerSubmittedToast = (props: Props) => {
-  const location = useLocation()
-  const { greaterThanSM } = useCustomMediaQuery()
-
-  useEffect(() => {
-    const isAnswersSubmitted = location.state?.isAnswersSubmitted || false
+export const useAnswerSubmittedToast = () => {
```
Comment (BamMironov) [inline src/entities/activity/model/hooks/useAnswerSubmittedToast.ts:3]: It looks like you can rename the hook to `useToast` as there is no logic related answer left

Comment (BamMironov) [reply src/entities/activity/model/hooks/useAnswerSubmittedToast.ts:3]: And it should be moved to the shared folder

**Review by BamMironov — CHANGES_REQUESTED** [review]

File: src/entities/activity/model/hooks/useAnswerSubmittedToast.ts (hunk 7 above)
Comment (moiskillnadne) [reply src/entities/activity/model/hooks/useAnswerSubmittedToast.ts:3]: Already done it in the next PR

**Review by BamMironov — APPROVED** [review]

**Pull request merged**